### PR TITLE
fix: remove all existence of serviceOptions

### PR DESCRIPTION
**Pull request by AllanZhengYP**
#### Description of changes

Remove extra layer of `serviceOptions` interface under `options` parameter in the Auth and Analytics API inputs. Thus the interface is aligned with Storage APIs

~~Confirmed no integration test app update needed~~

#### Description of how you validated changes
Unit test

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by stocaaro — DISMISSED** [review]

**Review by israx — DISMISSED** [review]
Thanks so much @AllanZhengYP  for putting this together. DX is so much better now!

**Review by ashwinkumar6 — APPROVED** [review]
Thanks for updating this @AllanZhengYP !

**Review by kvramyasri7 — DISMISSED** [review]
lgtm.

**Review by stocaaro — DISMISSED** [review]

**Review by israx — DISMISSED** [review]

File: packages/auth/src/types/options.ts [hunk 1]
```diff
@@ -6,7 +6,7 @@ import { AuthUserAttributes, AuthUserAttributeKey } from './models';
 /**
  * Base type for service options.
  */
-export type AuthServiceOptions = any;
+export type AuthServiceOptions = Record<string, any>;
```
Comment (cshfang) [inline packages/auth/src/types/options.ts:9]: Can we make this same change to Analytics/InApp/Push as part of this PR?

**Pull request merged**